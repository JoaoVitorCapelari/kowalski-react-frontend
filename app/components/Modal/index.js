/**
*
* Modal
*
*/

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

class Modal extends React.PureComponent {
  static propTypes = {
    children: PropTypes.any,
    active: PropTypes.bool,
    onDismiss: PropTypes.func,
  };

  componentDidMount() {
    document.addEventListener('keydown', this.props.onDismiss, false);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.props.onDismiss, false);
  }

  render() {
    const { children, active, onDismiss } = this.props;

    let classes = 'modal';
    if (active) {
      classes += ' is-active';
    }

    return (
      <div className={classes}>
        <div className="modal-background"></div>
        <div className="modal-content">
          { children }
        </div>
        <button className="modal-close is-large" aria-label="close" onClick={onDismiss}></button>
      </div>
    );
  }
}

export default Modal;
