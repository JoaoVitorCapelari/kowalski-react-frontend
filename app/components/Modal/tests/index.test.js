import React from 'react';
import { mount } from 'enzyme';

import Modal from '../index';

describe('<Modal />', () => {
  let modalText;
  let onDismiss;
  let active;
  let renderedComponent;

  beforeEach(() => {
    modalText = 'teste';
    onDismiss = jest.fn();
  });

  const renderComponent = () => mount(
    <Modal active={active} onDismiss={onDismiss}>
      { modalText }
    </Modal>
  );

  describe('when it is active', () => {
    beforeEach(() => {
      active = true;
      renderedComponent = renderComponent();
    });

    it('renders', () => {
      expect(renderedComponent.html()).toMatchSnapshot();
    });

    it('unmounts', () => {
      renderedComponent.unmount();
    });

    it('dismisses on esc key press', () => {
      renderedComponent.instance().dismissOnEsc({ keyCode: 10 });
      expect(onDismiss).not.toHaveBeenCalled();
      renderedComponent.instance().dismissOnEsc({ keyCode: 27 });
      expect(onDismiss).toHaveBeenCalled();
    });
  });

  describe('when it is not active', () => {
    beforeEach(() => {
      active = false;
      renderedComponent = renderComponent();
    });

    it('does not render', () => {
      expect(renderedComponent.html()).toMatchSnapshot();
    });
  });
});
