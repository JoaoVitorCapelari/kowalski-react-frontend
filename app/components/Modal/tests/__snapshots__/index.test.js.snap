// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> when it is active renders 1`] = `"<div class=\\"modal is-active\\"><div class=\\"modal-background\\"></div><div class=\\"modal-content\\">teste</div><button class=\\"modal-close is-large\\" aria-label=\\"close\\"></button></div>"`;

exports[`<Modal /> when it is not active does not render 1`] = `"<div class=\\"modal\\"><div class=\\"modal-background\\"></div><div class=\\"modal-content\\">teste</div><button class=\\"modal-close is-large\\" aria-label=\\"close\\"></button></div>"`;
